portfolio/src/main/webapp/piano_im.js

```diff
@@ -0,0 +1,49 @@
+const notes = new Map();
+notes.set('C', 'white');
+notes.set('C#', 'black');
+notes.set('D', 'white');
+notes.set('D#', 'black');
+notes.set('E', 'white');
+notes.set('F', 'white');
+notes.set('F#', 'black');
+notes.set('G', 'white');
+notes.set('G#', 'black');
+notes.set('A', 'white');
+notes.set('A#', 'black');
+notes.set('B', 'white');
+notes.set('C2', 'white');
+notes.set('C#2', 'black');
+notes.set('D2', 'white');
+notes.set('D#2', 'black');
+notes.set('E2', 'white');
+
+const svgns = 'http://www.w3.org/2000/svg';
+const svg = document.getElementsByTagName('svg')[0];
+const whiteWidth = 100;
+const blackWidth = 30;
+const whiteHeight = '300';
+const blackHeight = '200';
+let x = 0;
+y = '0';
+for (const name of notes.keys()) {
+  const rect = document.createElementNS(svgns, 'rect');
+  rect.setAttributeNS(null, 'x', x);
+  rect.setAttributeNS(null, 'y', y);
+  rect.setAttributeNS(null, 'data-key', name);
+  if (notes.get(name) == 'white') {
```

Comment:
nit: this big if/else can be a bit simplified by moving common statements outside

```
let width = 0;
let height = 0;
if (notes.get(name) == 'white') {
  width = whiteWidth;
  height = whiteHeight;
  rect.setAttributeNS(null, 'stroke', 'black');
  rect.setAttributeNS(null, 'stroke-width', '5');
} else {
  width = blackWidth;
  height = blackHeight;
}
rect.setAttributeNS(null, 'height', height);
rect.setAttributeNS(null, 'width', width);
rect.setAttributeNS(null, 'fill', notes.get(name));
rect.setAttributeNS(null, 'class', 'key');
x += width;
svg.appendChild(rect);
```